### ISSUE TITLE
Development and debug environmental setup

### ISSUE THREAD

**Issue by Adam-Langley** [brutella/hkcam#79]
Hi @brutella ,

I am interesting in adapting your code to integrate gstreamer on Raspberry pi.
To get up and running, I understand I need Delve to debug Go on the Pi.
I've also found that Delve will not run on Raspbian given its 32bitARM nature.

Can I ask, whats your development environment for Raspberry Pi debugging? Can you provide some pointers on getting this set up?

My ideal scenario is VS Code - remote debugging to a Pi4 on Raspbian.

I say Raspbian, because I have a lot of knowledge there - I'd prefer not to have to switch to Ubuntu for example.

Thanks

**Comment by brutella** [comment]
I actually do very little debugging on the Raspberry Pi itself. 
I'm developing hkcam on a Mac, where it works with the built-in camera.

**Comment by Adam-Langley** [comment]
Thanks for getting back @brutella  - well done on getting it to work on a Pi in that case!
I had a crack at it - Delve requires a 64bit O/S, so no chance of debugging on Raspbian while it's still 32-bit only. So I resorted to Ubuntu, and managed to get a golang remote debugging situation up and running between VS code on my Mac and Ubunu Server 18 on the Pi, however I realised that was only part of the problem, as I still wanted to be using OpenMax/MMAL for the encoding side of things - which is only available for 32-bit ARM... ARGH!
So - I'm back on Raspbian again - gonna have to use a Python based HAP library.
But it was a nice foray into Go using your library, while it lasted :)
Thanks and keep up the tinkering!
